### ISSUE TITLE
[8.0.0-beta.2, svelte-table] Custom cell components not working in dev mode

### ISSUE THREAD

**Issue by heikkilamarko** [TanStack/table#3971]
### Describe the bug

I'm working with a SvelteKit application (SvelteKit v1.0.0-next.345).

The following check in `svelte-table` package is not working correctly in `dev` mode:

```
function isSvelteClientComponent(component: any) {
  return component.prototype instanceof SvelteComponent
}
```

In `dev` mode the `component` is of type `ProxyComponent` from `svelte-hmr` package:

https://github.com/sveltejs/svelte-hmr/blob/ac0c5bfb908a7c6ebe385997c65facccdcf9a167/packages/svelte-hmr/runtime/proxy.js#L125

The check is working correctly in `prod` builds.

### Your minimal, reproducible example

See "Steps to reproduce"

### Steps to reproduce

You can reproduce the problem by adding the following code to your app, where `ValueCell` is a normal Svelte component (`ValueCell.svelte`). Then running your app in dev mode (`npm run dev`).

```
table.createDataColumn('value', {
  header: 'Value',
  cell: ValueCell
})
```

The following error appears in the browser console:

<img width="2034" alt="screenshot" src="https://user-images.githubusercontent.com/3757314/170829220-fd4534c2-13cb-4db2-99d1-bff57ea8f00a.png">

### Expected behavior

Custom cell components (Svelte components) render correctly in `dev` similar to `prod`.

### How often does this bug happen?

Every time

### Platform

macOS, all browsers

### react-table version

8.0.0-beta.2

### Terms & Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I understand that if my bug cannot be reliable reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by heikkilamarko** [comment]
As a temporary workaround I added the following `startsWith`check:

```
function isSvelteClientComponent(component) {
  return component.prototype instanceof SvelteComponent || component.name?.startsWith("Proxy<");
}
```

**Comment by tannerlinsley** [comment]
@Mokshit06 Any ideas here?

**Comment by KevinVandy** [comment]
Possibly relevant conversation from last month

<img width="1083" alt="image" src="https://user-images.githubusercontent.com/28243511/170920850-fd75a252-1474-41f9-8320-7a68b7724f71.png">

**Comment by Mokshit06** [comment]
The workaround by @heikkilamarko is actually the way recommended by svelte-kit to check if it's an hmr component (https://github.com/sveltejs/svelte-hmr/blob/ac0c5bfb908a7c6ebe385997c65facccdcf9a167/packages/svelte-hmr/runtime/proxy.js#L343). I'll add a few more checks like checking if the window has `__SVELTE_HMR` to ensure that it's in an hmr environment and create a pr

**Comment by KevinVandy** [comment]
Has this been solved in the new beta?

**Comment by Mokshit06** [comment]
yep this has been solved in #3980
